### PR TITLE
Deserialize IIR

### PR DESCRIPTION
**Pull request by lukasm91**
See https://github.com/MeteoSwiss-APN/dawn/pull/236

### CODE REVIEW

**Comment by lukasm91** [comment]
launch pipeline

**Comment by lukasm91** [comment]
launch pipeline

File: src/gtclang/Frontend/GTClangASTConsumer.cpp [hunk 1]
```diff
@@ -229,9 +231,16 @@ void GTClangASTConsumer::HandleTranslationUnit(clang::ASTContext& ASTContext) {
...
                            ? "<restored>"
-                           : stencilPair.second->Name)))
+                           : stencilPair.second->Name))) {
         context_->getDiagnostics().report(Diagnostics::err_fs_error) << dawn::format(
             "unable to replace stencil code at: %s", stencilDecl->getLocation().printToString(SM));
+      } else {
+        num_stencils_generated++;
+      }
+      if(context_->getOptions().DeserializeIIR != "" && num_stencils_generated > 1) {
+        printf("WARNING: more than one stencil generated but only one stencil deserialized from IIR!\n");
```
Comment (Stagno) [inline src/gtclang/Frontend/GTClangASTConsumer.cpp]: No need for this i guess

File: src/gtclang/Frontend/GlobalVariableParser.cpp [hunk 2]
```diff
@@ -149,16 +149,29 @@ void GlobalVariableParser::parseGlobals(clang::CXXRecordDecl* recordDecl) {
...
-        value->setValue((double)std::atof(valueStr.c_str()));
+      // this slightly unelegant procedure is needed since FloatingLiteral does not cast from
+      // expressions without trailing do (e.g. `12.' would cast, `12' wouldn't.)
+      } else if((dyn_cast<FloatingLiteral>(init) != nullptr || dyn_cast<IntegerLiteral>(init) != nullptr)
+          && typeKind == dawn::sir::Value::Double) {
+        IntegerLiteral* il = dyn_cast<IntegerLiteral>(init);
+        FloatingLiteral* fl = dyn_cast<FloatingLiteral>(init);
+        std::string valueStr;
+        if (fl != nullptr) {
+          llvm::SmallVector<char, 10> valueVec;
```
Comment (Stagno) [inline src/gtclang/Frontend/GlobalVariableParser.cpp]: just a small improvement suggestion: putting magic number 10 into a const int member of GlobalVariableParser and maybe making it a 20 (3.1415926535 breaks it already).

File: src/gtclang/Frontend/GTClangASTConsumer.cpp (hunk 1 above)
Comment (Stagno) [reply src/gtclang/Frontend/GTClangASTConsumer.cpp]: update: change to std::cerr << ...

File: src/gtclang/Frontend/GlobalVariableParser.cpp (hunk 2 above)
Comment (Stagno) [reply src/gtclang/Frontend/GlobalVariableParser.cpp]: update: for now we keep it, but plan to refactor magic numbers away from the whole code sometime

**Comment by lukasm91** [comment]
launch pipeline

**Comment by mroethlin** [comment]
launch pipeline

**Pull request merged**